src/coreclr/tools/Common/TypeSystem/IL/ILStackHelper.cs

```diff
@@ -284,12 +248,12 @@ public static int ComputeMaxStack(this MethodIL methodIL)
                     case ILOpcode.callvirt:
                     case ILOpcode.newobj:
                         {
-                            int token = ReadILToken(ilbytes, currentOffset + 1);
-                            object obj = methodIL.GetObject(token);
-                            MethodSignature sig = obj is MethodSignature ?
-                                (MethodSignature)obj :
-                                ((MethodDesc)obj).Signature;
-                            int adjustment = sig.Length;
+                            object obj = methodIL.GetObject(ilReader.ReadILToken());
+                            hasReadOperand = true;
+
+                            MethodSignature sig = obj is MethodSignature methodSignature ?
+                                methodSignature : ((MethodDesc)obj).Signature;
+                            short adjustment = (short)sig.Length;
```

Comment:
There is no 16-bit limit on signature length. This cast can overflow silently.

Reply:
Came across my mind too for a moment. Reverted back to int, [reverted.](https://github.com/dotnet/runtime/pull/97794/commits/d6b2b334eb09f44f2e44e76b4157c1b003b9d98a)